buildhat/motors.py

```diff
@@ -217,16 +244,31 @@ def start(self, speed=None):
 
         :param speed: Speed ranging from -100 to 100
         """
+        if self.runmode == Motor.runmodes['Free']:
+            if self.current_freerun_speed == speed:
+                # Already running at this speed, do nothing
```

Comment:
That is a first order filter. 
In a scenario with a constantly changing speed value, this code will fail to reduce the message rate. Therefore, the interface must sill be able to sustain the full message speed in order to rule out a random failure.

Reply:
True, but but the Pi is fast compared to the serial port and you're going to need/want all the bandwidth you can get out of the serial port in your scenario. This is seems kind of cheap.

Reply:
Agreed, in this case there might be a minor benefit.